File: idf/tst/idfsample1.f
       program couette  
c  The first sample program in FORTRAN  

c    initialization of input IDF data file String  
       character*(*)  Fname 
       integer Flen 
       parameter (Flen = 11 , Fname = 'couette.dat')

c    declaration of input parameters  
       real*8  Lz,Pr(3)  
       real*4  Knu,Uw,Tw  
       integer*4  Nz,Nv(2)  
       integer*4  ier  

c    start the IDF package  
          ier=idfinit(0)  
       if(ier.ne.0) go to 1  

c    open the data file  
          ier=idfopen(Fname,Flen)  
       if(ier.ne.0) go to 2  

c    reading the data from file  
          ier=idfd('Lz', 2, Lz)  
       if(ier.ne.0)  go to  2  
          ier=idff('Knu', 3, Knu)  
       if(ier.ne.0)  go  to  2  
          ier=idff('Uw', 2, Uw)  
       if(ier.ne.0)  go  to  2  
          ier=idff('Tw', 2,  Tw)  
       if(ier.ne.0)  go  to  2  
          ier=idfdarr('Pr', 2, Pr, 3)  
       if(ier.ne.3)  go  to  2  
          ier=idfi('Nz', 2, Nz)  
       if(ier.ne.0)  go  to  2  
          ier=idfiarr('Nv', 2, Nv, 2)  
       if(ier.ne.2)  go  to  2  

c    print the imported data sets  
       print 10,Lz,Knu  
 10    format('Lz=',E12.3, 'Knu=',E12.3)  
       print 20,Uw,Tw  
 20    format('Uw=',E12.3, '  Tw=',E12.3)  
       print 30,Pr(1),Pr(2),Pr(3)  
 30    format ('Pr=',E12.3, ',', E12.3, ',', E12.3)  
       print 40,Nz,Nv(1),Nv(2)  
 40    format('Nz=',I5, 'Nv=',I5,',', I5)  

c    close the data file  
 2     call idfclose  

c    finish the IDF  
 1     call idfinish  
       stop  
       end  

File: idf/tst/idfsample1.c
#include <stdio.h>

#define IDF_CPP 0
#define IDF_CPP_STYLE 1
#define IDF_FORTRAN_TYPE 3
#include "idfuser.h"



int main()
/* The first sample program in C (see IDF manual) */
{
/*initialization of input IDF data file String*/
static char Fname[12] = "couette.dat";

/* declaration of input parameters */
double Lz,Pr[3];
float Knu,Uw,Tw;
int Nz,Nv[2];
int ier;

/* start the IDF package */
   ier=idf_init();
if(ier) goto er;

/* open the data file */
   ier=idf_open(Fname);
if(ier) goto err;

/* reading the data from file */
   ier=idf_d   ("Lz" ,    &Lz );
if(ier) goto err;
   ier=idf_f   ("Knu",    &Knu);
if(ier) goto err;
   ier=idf_f   ("Uw" ,    &Uw );
if(ier) goto err;
   ier=idf_f   ("Tw" ,    &Tw );
if(ier) goto err;
   ier=idf_darr("Pr" ,  Pr , 3  );
if(ier!=3) goto err;
   ier=idf_i   ("Nz" ,    &Nz );
if(ier) goto err;
   ier=idf_iarr("Nv" ,  Nv , 2 );
if(ier!=2) goto err;

/* print the imported data sets*/
printf("Lz=%12.3e Knu=%12.3e\n",   Lz,Knu);
printf("Uw=%12.3e  Tw=%12.3e\n",   Uw,Tw);
printf("Pr=%12.3e %12.3e %12.3e\n",Pr[0],Pr[1],Pr[2]);
printf("Nz=%5d Nv=%5d, %5d\n",     Nz,Nv[0],Nv[1]);

/* close the data file */
err: idf_close();

/* finish the IDF */
er: idf_finish();

return ier;
}
